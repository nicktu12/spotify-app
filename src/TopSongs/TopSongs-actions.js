export const loadSongsAction = (token) => {
  return {
    type: 'LOAD_SONGS',
    token
  };
};

export const loadSongsShortTerm = (token) => {
  return {
    type: 'LOAD_SONGS_SHORT_TERM',
    token
  };
};

export const loadSongsAllTime = (token) => {
  return {
    type: 'LOAD_SONGS_ALL_TIME',
    token
  };
};

export const postPlaylist = (token, id, array) => {
  return {
    type: 'POST_PLAYLIST',
    payload: {token, id, array}
  }
}
