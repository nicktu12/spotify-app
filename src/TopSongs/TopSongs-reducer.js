export function topSongs(state=[], action){
  switch (action.type) {
  case 'TOP_SONGS':
    return action.topSongs;
  default:
    return state;
  }
}

export function postPlaylist(state=[], action){
  switch (action.type) {
  case 'POST_PLAYLIST_SUCCESS':
    return action.playlistId;
  default:
    return state;
  }
}

