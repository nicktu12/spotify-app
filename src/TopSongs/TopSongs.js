import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Meter } from '../Meter/Meter';
import { 
  loadSongsAction, 
  loadSongsShortTerm, 
  loadSongsAllTime, 
  postPlaylist,
} from './TopSongs-actions';

export class TopSongs extends React.Component{

  constructor(){
    super();

    this.state = {
      selected: null,
    };
  } 

  componentDidMount(){
    if (!this.props.accessToken.length) {
      this.props.history.push('/login');
      return;
    }
    if (this.props.match.path === '/top40/month') {
      return this.props.topSongsShortTerm.length ?
        null :
        this.props.loadSongsShortTerm(this.props.accessToken);
    }
    if (this.props.match.path === '/top40') {
      return this.props.topSongs.length ?
        null : 
        this.props.loadSongs(this.props.accessToken); 
    }
    if (this.props.match.path === '/top40/alltime') {
      return this.props.topSongsAllTime.length ?
        null : 
        this.props.loadSongsAllTime(this.props.accessToken); 
    }
  }

  renderSongs = () => {
    if (this.props.match.path === '/top40') {
      return this.props.topSongs.map((song, index) => (
        <li 
          key={'top songs ' + index}
          className={
            this.state.selected === index ?
              'selected songs' :
              'unselected songs' }
        >
          <button onClick={()=>this.addSelected(index)}>
            <span>{song.title}</span> {song.artists}
          </button>
        </li>
      ));
    }
    if (this.props.match.path === '/top40/month') {
      return this.props.topSongsShortTerm.map((song, index) => (
        <li key={'top songs ' + index}>
          <button onClick={()=>this.addSelected(index)}>
            <span>{song.title}</span> {song.artists}
          </button>
        </li>
      ));
    }
    if (this.props.match.path === '/top40/alltime') {
      return this.props.topSongsAllTime.map((song, index) => (
        <li key={'top songs ' + index}>
          <button onClick={()=>this.addSelected(index)}>
            <span>{song.title}</span> {song.artists}
          </button>
        </li>
      ));
    }
  }

  addSelected = (index) => (
    this.state.selected === null ? 
      this.setState({ selected: index}) : 
      this.updateSelected(index)
  )

  updateSelected = (index) => {
    this.state.selected === index ?
      this.setState({ selected: null }) :
      this.setState({ selected: index });
  }


  showLoading = () => { 
    if (this.props.match.path === '/top40/month') {
      return this.props.topSongsShortTerm.length ?
        null :
        <img src={require('../Assets/bars.svg')} alt="loading icon" />;
    }
    if (this.props.match.path === '/top40') {
      return this.props.topSongs.length ?
        null : 
        <img src={require('../Assets/bars.svg')} alt="loading icon" />;
    }
    if (this.props.match.path === '/top40/alltime') {
      return this.props.topSongsAllTime.length ?
        null : 
        <img src={require('../Assets/bars.svg')} alt="loading icon" />;
    }
  }

  determineClass = (path) =>  {
    if (path === null) {
      return window.location.href === "http://localhost:3000/top40";
    }
    return window.location.href.includes(path);
  }

  renderInfoCard = (info) => (
    info === undefined ?
      <section>
        <h4>{this.props.userInfo.name}</h4>
        {
          this.props.userInfo.image &&
          <img src={this.props.userInfo.image} alt='user' />
        }
        <p className='user-info'>
          {
            this.props.userInfo.followers &&
            <span>
              <span>Followers:</span> <span className='alt-text'>
                {this.props.userInfo.followers}
              </span>
            </span>
          }
        </p>
      </section>
      :
      <section>
        <h4>{info.title}</h4>
        <img src={info.image} alt={info.album + 'album art'} />
        <div>
          <p>
            <span>Album:</span> <span className='alt-text'>
              {info.album}
            </span>
          </p>
          <p>
            <span>Artist:</span> <span className='alt-text'>
              {info.artists}
            </span>
          </p>
          <p>
            <span>Popularity:</span> <span className='alt-text'>
              <Meter percent={info.popularity / 100} rounded={true} />
            </span>
          </p>
        </div>
      </section>
  )

  renderInfoCardSwitch = (path) => {
    if (path === '/top40') {
      return this.renderInfoCard(this.props.topSongs[this.state.selected])
    }
    if (path === '/top40/month') {
      return this.renderInfoCard(this.props.topSongsShortTerm[this.state.selected])
    }
    if (path === '/top40/alltime') {
      return this.renderInfoCard(this.props.topSongsAllTime[this.state.selected])
    }
  }

  getSongUriArray = songs => songs.map(song=>(song.uri));

  disableBtn = () => this.refs.btn.setAttribute('disabled', 'disabled');

  render(){
    return (   
      <div className='playlist-div'>
        { this.renderInfoCardSwitch(this.props.match.path) }
        <h2>
          {this.showLoading()} Top 40 {this.showLoading()}
          <div>
            <button 
              className={
                this.determineClass('month') ? 
                  'playlist-path-active' : 
                  null
              } 
              onClick={()=>this.props.history.push('/top40/month')}>
              This month
            </button>
            <button 
              className={
                this.determineClass(null) ? 
                  'playlist-path-active' : 
                  null
              } 
              onClick={()=>this.props.history.push('/top40')}>
               This year
            </button>
            <button 
              className={
                this.determineClass('alltime') ? 
                  'playlist-path-active' : 
                  null
              } 
              onClick={()=>this.props.history.push('/top40/alltime')}>
                All Time
            </button>
          </div>
        </h2>
        <button 
          ref="btn"
          onClick={()=>{
            this.props.postPlaylist(this.props.accessToken, this.props.userInfo.id, this.getSongUriArray(this.props.topSongs))
            this.disableBtn()
          }}
          className={this.props.match.path === '/top40' ? 'post-btn display-post-btn' : 'post-btn'}
        >
          Add to Spotify
        </button>
        <ol>
          {this.props.topSongs && this.renderSongs()}
        </ol>
      </div>
    );
  }
}

const mapStateToProps = store => ({
  accessToken: store.accessToken,
  topSongs: store.topSongs,
  topSongsShortTerm: store.topSongsShortTerm,
  topSongsAllTime: store.topSongsAllTime,
  userInfo: store.userInfo,
});

const mapDispatchToProps = dispatch => {
  return {
    loadSongs: (token) => dispatch(loadSongsAction(token)),
    loadSongsShortTerm: (token) => dispatch(loadSongsShortTerm(token)),
    loadSongsAllTime: (token) => dispatch(loadSongsAllTime(token)),
    postPlaylist: (token, id, array) => dispatch(postPlaylist(token, id, array))
  };
};

TopSongs.propTypes = {
  accessToken: PropTypes.string,
  match: PropTypes.object,
  topSongsShortTerm: PropTypes.arrayOf(PropTypes.object),
  topSongs: PropTypes.arrayOf(PropTypes.object),
  topSongsAllTime: PropTypes.arrayOf(PropTypes.object),
  loadSongs: PropTypes.func,
  loadSongsShortTerm: PropTypes.func,
  loadSongsAllTime: PropTypes.func,
  history: PropTypes.oneOfType([PropTypes.object]),
  userInfo: PropTypes.object,
};

export default connect(mapStateToProps, mapDispatchToProps)(TopSongs);
