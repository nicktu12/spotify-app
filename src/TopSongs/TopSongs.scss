@import '../Styles/Variables.scss';

.playlist-div {
	height: calc(100vh - 90px);
	box-sizing: border-box;
	border-radius: 0px 0px $border-rad $border-rad;
}

.playlist-div h2 {
  background-color: $pop-accent-o;
  color: $white;
  text-align: center;
  padding-top: 10px;
	height: 91px;
	box-sizing: border-box;
	font-family: $header-font;
	font-size: 32px;
}

h2 div button {
  background-color: rgba(0,0,0,0);
  border: none;
  display: inline-block;
  font-family: $body-font;
  color: $white;
  vertical-align: middle;
  font-size: 16px;
  padding: 0px 15px;
  transition: all 200ms ease-in;
}

h2 div button:hover {
  cursor: pointer;
  color: $dark-bg;
  transition: all 200ms ease-in;
}

.playlist-path-active {
  color: $dark-bg;
}

.playlist-div ol {
  background-color: $dark-bg;
  color: $light-3;
  text-align: left;
  padding-left: calc(30vw + 35px);
	overflow: scroll;
	height: calc(99vh - 174px);
}

.playlist-div li {
 font-family: $header-font; 
 border-radius: 1px;
 box-sizing: border-box;
 height: 42px;
 margin: 4px 0px;
}

.playlist-div li:hover {
  background-color: $dark;
  box-sizing: border-box;
  border-bottom: 2px solid $pop-accent;
}

.playlist-div li:hover button {
  color: #E1B484;
}

.selected button {
  color: #E1B484;
}

li span {
	color: $white;
	font-size: 20px;
	font-family: $header-font;
  padding: 0px 3px 0px 10px;
}

.post-btn {
  display: none;
}

.display-post-btn {
  display: initial;
  position: absolute;
  right: 25px;
  top: 206px;
  background-color: $spotify-green;
  border: none;
  color: $white;
  font-size: 16px;
  font-family: $header-font;
  outline-color: $spotify-green;
  padding: 2px 30px;
}
