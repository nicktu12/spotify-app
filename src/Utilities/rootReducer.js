import { combineReducers } from 'redux';
import { accessToken, topArtists, userInfo } from '../Login/Login-reducer';
import { topSongs, topSongsShortTerm, topSongsAllTime } from '../Playlist/Playlist-reducer';

export default combineReducers({
  accessToken, 
  topArtists,
  topSongs,
  topSongsShortTerm,
  topSongsAllTime,
  userInfo,
});
