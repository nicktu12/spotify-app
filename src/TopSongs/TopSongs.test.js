import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { TopSongs } from './TopSongs';
import { mockTopSongs, mockUserInfo, mockAccessToken } from '../Utilities/mockData';

describe('TopSongs component tests', ()=>{
  const mockFunc = jest.fn();
  const wrapper = shallow(
    <TopSongs 
      accessToken={mockAccessToken}
      match={{path: '/top40'}}
      topSongs={mockTopSongs}
      userInfo={mockUserInfo}
    />
  );

  it('should match snapshot render', ()=>{
    expect(wrapper).toMatchSnapshot();
  })
  
  it('should have a default state', ()=>{
    expect(wrapper.state().selected).toEqual(null);
  })

  it('should render top songs',()=>{
    expect(wrapper.instance().renderSongs(mockTopSongs).length).toEqual(3);
  })

  it('should be able to add a selected card to state by index', ()=>{
    wrapper.instance().addSelected(1);

    expect(wrapper.state().selected).toEqual(1);
  })

  it('should be able to update selected card if a new card is clicked', ()=>{
    wrapper.instance().addSelected(1);
    wrapper.instance().addSelected(3);

    expect(wrapper.state().selected).toEqual(3);
  }) 

  it('should be able to return state to null if same card is clicked', ()=>{
    wrapper.instance().addSelected(1);
    wrapper.instance().addSelected(1);

    expect(wrapper.state().selected).toEqual(null);
  })

  it('should show two loading svgs if no top songs exist', ()=>{
    const noSongsWrapper = shallow(
      <TopSongs 
        loadSongs={mockFunc}
        match={{path: '/top40'}}
        authCodeToSagas={mockFunc}
        topSongs={[]}
        userInfo={mockUserInfo}
        accessToken={mockAccessToken}
      />
    );

    expect(noSongsWrapper.find('h2').find('img').length).toEqual(2)
  })

  it('should show user info if undefined is passed to renderInfoCard function', ()=>{
    expect(wrapper.instance().renderInfoCard(undefined).props.children[0].props.children).toEqual('Edgar the Cat')
    expect(wrapper.instance().renderInfoCard(mockTopSongs[0]).props.children[0].props.children).toEqual('In Da Club')
  })

  it('should show artist info if passed artist information',()=>{
    expect(wrapper.instance().renderInfoCard(mockTopSongs[0]).props.children[0].props.children).toEqual('In Da Club')
  })

});
