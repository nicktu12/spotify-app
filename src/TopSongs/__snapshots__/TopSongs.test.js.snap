// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSongs component tests should match snapshot render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopSongs
    accessToken="BxxxxxxxU8xxxxxxxvK5xxxxxxxxxsBrtbsaM_37uIB1yaFCPApS0Iczif6KIKq8EXAwXtoQyG8FduappKkTi0JYNO-a4E7ioKBNlpMmCI9zf4KiMd9_ZN3YpFXv0TRUhCuxq_WXHLoRn-M-cSPJ5AYZdRFrB7_dwkwtLsOr"
    match={
      Object {
        "path": "/top40",
      }
    }
    topSongs={
      Array [
        Object {
          "album": "Half a Dollar",
          "artists": Array [
            "50 Cent",
          ],
          "image": "https://i.scdn.co/imaaeaabaaaaaaacaff4a0d7a879b265ef692b7307ea50",
          "popularity": 88,
          "title": "In Da Club",
          "uri": "spotify:trackt6tCtLttcttt5HttttJgUBA",
        },
        Object {
          "album": "A Seat at the Table",
          "artists": Array [
            "Solange",
          ],
          "image": "https://i.scdn.co/jdhshjdfgjhgsfdj",
          "popularity": 77,
          "title": "Cranes In The Sky",
          "uri": "spotify.track34534hj34jh4",
        },
        Object {
          "album": "Shigeto",
          "artists": Array [
            "Shigeto",
          ],
          "image": "https://i.scden.com/asdfasd",
          "popularity": 54,
          "title": "Huron River Drive",
          "uri": "spotify.track55h5j5j5",
        },
      ]
    }
    userInfo={
      Object {
        "email": "edgarthecat@cats.com",
        "image": "https://scontent.xx.fbcdn.net/v/t1.0-1/aaaaaaaaaaaaaa558a1aaaaa0aaaaaa5063aaaa4aaaa12218136501_n.jpg?oh=53ef504a0ba139726593f4497886fa16&oe=5A721441",
        "name": "Edgar the Cat",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section>
          <h4>
            Edgar the Cat
          </h4>
          <img
            alt="user"
            src="https://scontent.xx.fbcdn.net/v/t1.0-1/aaaaaaaaaaaaaa558a1aaaaa0aaaaaa5063aaaa4aaaa12218136501_n.jpg?oh=53ef504a0ba139726593f4497886fa16&oe=5A721441"
          />
          <p
            className="user-info"
          />
          <p />
        </section>,
        <h2>
           Top 40 
          <div>
            <button
              className={null}
              onClick={[Function]}
            >
              This month
            </button>
            <button
              className={null}
              onClick={[Function]}
            >
              This year
            </button>
            <button
              className={null}
              onClick={[Function]}
            >
              All Time
            </button>
          </div>
        </h2>,
        <button
          className="post-btn display-post-btn"
          onClick={[Function]}
        >
          Add to Spotify
        </button>,
        <ol>
          <li
            className="unselected songs"
          >
            <button
              onClick={[Function]}
            >
              <span>
                In Da Club
              </span>
               
              50 Cent
            </button>
          </li>
          <li
            className="unselected songs"
          >
            <button
              onClick={[Function]}
            >
              <span>
                Cranes In The Sky
              </span>
               
              Solange
            </button>
          </li>
          <li
            className="unselected songs"
          >
            <button
              onClick={[Function]}
            >
              <span>
                Huron River Drive
              </span>
               
              Shigeto
            </button>
          </li>
        </ol>,
      ],
      "className": "playlist-div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Edgar the Cat
            </h4>,
            <img
              alt="user"
              src="https://scontent.xx.fbcdn.net/v/t1.0-1/aaaaaaaaaaaaaa558a1aaaaa0aaaaaa5063aaaa4aaaa12218136501_n.jpg?oh=53ef504a0ba139726593f4497886fa16&oe=5A721441"
            />,
            <p
              className="user-info"
            />,
            <p />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edgar the Cat",
            },
            "ref": null,
            "rendered": "Edgar the Cat",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "user",
              "src": "https://scontent.xx.fbcdn.net/v/t1.0-1/aaaaaaaaaaaaaa558a1aaaaa0aaaaaa5063aaaa4aaaa12218136501_n.jpg?oh=53ef504a0ba139726593f4497886fa16&oe=5A721441",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "user-info",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            " Top 40 ",
            null,
            <div>
              <button
                className={null}
                onClick={[Function]}
              >
                This month
              </button>
              <button
                className={null}
                onClick={[Function]}
              >
                This year
              </button>
              <button
                className={null}
                onClick={[Function]}
              >
                All Time
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          null,
          " Top 40 ",
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className={null}
                  onClick={[Function]}
                >
                  This month
                </button>,
                <button
                  className={null}
                  onClick={[Function]}
                >
                  This year
                </button>,
                <button
                  className={null}
                  onClick={[Function]}
                >
                  All Time
                </button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This month",
                  "className": null,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "This month",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This year",
                  "className": null,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "This year",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "All Time",
                  "className": null,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "All Time",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add to Spotify",
          "className": "post-btn display-post-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add to Spotify",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="unselected songs"
            >
              <button
                onClick={[Function]}
              >
                <span>
                  In Da Club
                </span>
                 
                50 Cent
              </button>
            </li>,
            <li
              className="unselected songs"
            >
              <button
                onClick={[Function]}
              >
                <span>
                  Cranes In The Sky
                </span>
                 
                Solange
              </button>
            </li>,
            <li
              className="unselected songs"
            >
              <button
                onClick={[Function]}
              >
                <span>
                  Huron River Drive
                </span>
                 
                Shigeto
              </button>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "top songs 0",
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={[Function]}
              >
                <span>
                  In Da Club
                </span>
                 
                50 Cent
              </button>,
              "className": "unselected songs",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    In Da Club
                  </span>,
                  " ",
                  Array [
                    "50 Cent",
                  ],
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "In Da Club",
                  },
                  "ref": null,
                  "rendered": "In Da Club",
                  "type": "span",
                },
                " ",
                "50 Cent",
              ],
              "type": "button",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "top songs 1",
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={[Function]}
              >
                <span>
                  Cranes In The Sky
                </span>
                 
                Solange
              </button>,
              "className": "unselected songs",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Cranes In The Sky
                  </span>,
                  " ",
                  Array [
                    "Solange",
                  ],
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cranes In The Sky",
                  },
                  "ref": null,
                  "rendered": "Cranes In The Sky",
                  "type": "span",
                },
                " ",
                "Solange",
              ],
              "type": "button",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "top songs 2",
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={[Function]}
              >
                <span>
                  Huron River Drive
                </span>
                 
                Shigeto
              </button>,
              "className": "unselected songs",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Huron River Drive
                  </span>,
                  " ",
                  Array [
                    "Shigeto",
                  ],
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Huron River Drive",
                  },
                  "ref": null,
                  "rendered": "Huron River Drive",
                  "type": "span",
                },
                " ",
                "Shigeto",
              ],
              "type": "button",
            },
            "type": "li",
          },
        ],
        "type": "ol",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section>
            <h4>
              Edgar the Cat
            </h4>
            <img
              alt="user"
              src="https://scontent.xx.fbcdn.net/v/t1.0-1/aaaaaaaaaaaaaa558a1aaaaa0aaaaaa5063aaaa4aaaa12218136501_n.jpg?oh=53ef504a0ba139726593f4497886fa16&oe=5A721441"
            />
            <p
              className="user-info"
            />
            <p />
          </section>,
          <h2>
             Top 40 
            <div>
              <button
                className={null}
                onClick={[Function]}
              >
                This month
              </button>
              <button
                className={null}
                onClick={[Function]}
              >
                This year
              </button>
              <button
                className={null}
                onClick={[Function]}
              >
                All Time
              </button>
            </div>
          </h2>,
          <button
            className="post-btn display-post-btn"
            onClick={[Function]}
          >
            Add to Spotify
          </button>,
          <ol>
            <li
              className="unselected songs"
            >
              <button
                onClick={[Function]}
              >
                <span>
                  In Da Club
                </span>
                 
                50 Cent
              </button>
            </li>
            <li
              className="unselected songs"
            >
              <button
                onClick={[Function]}
              >
                <span>
                  Cranes In The Sky
                </span>
                 
                Solange
              </button>
            </li>
            <li
              className="unselected songs"
            >
              <button
                onClick={[Function]}
              >
                <span>
                  Huron River Drive
                </span>
                 
                Shigeto
              </button>
            </li>
          </ol>,
        ],
        "className": "playlist-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Edgar the Cat
              </h4>,
              <img
                alt="user"
                src="https://scontent.xx.fbcdn.net/v/t1.0-1/aaaaaaaaaaaaaa558a1aaaaa0aaaaaa5063aaaa4aaaa12218136501_n.jpg?oh=53ef504a0ba139726593f4497886fa16&oe=5A721441"
              />,
              <p
                className="user-info"
              />,
              <p />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edgar the Cat",
              },
              "ref": null,
              "rendered": "Edgar the Cat",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "user",
                "src": "https://scontent.xx.fbcdn.net/v/t1.0-1/aaaaaaaaaaaaaa558a1aaaaa0aaaaaa5063aaaa4aaaa12218136501_n.jpg?oh=53ef504a0ba139726593f4497886fa16&oe=5A721441",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "user-info",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              " Top 40 ",
              null,
              <div>
                <button
                  className={null}
                  onClick={[Function]}
                >
                  This month
                </button>
                <button
                  className={null}
                  onClick={[Function]}
                >
                  This year
                </button>
                <button
                  className={null}
                  onClick={[Function]}
                >
                  All Time
                </button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            null,
            " Top 40 ",
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className={null}
                    onClick={[Function]}
                  >
                    This month
                  </button>,
                  <button
                    className={null}
                    onClick={[Function]}
                  >
                    This year
                  </button>,
                  <button
                    className={null}
                    onClick={[Function]}
                  >
                    All Time
                  </button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "This month",
                    "className": null,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "This month",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "This year",
                    "className": null,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "This year",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "All Time",
                    "className": null,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "All Time",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add to Spotify",
            "className": "post-btn display-post-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add to Spotify",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="unselected songs"
              >
                <button
                  onClick={[Function]}
                >
                  <span>
                    In Da Club
                  </span>
                   
                  50 Cent
                </button>
              </li>,
              <li
                className="unselected songs"
              >
                <button
                  onClick={[Function]}
                >
                  <span>
                    Cranes In The Sky
                  </span>
                   
                  Solange
                </button>
              </li>,
              <li
                className="unselected songs"
              >
                <button
                  onClick={[Function]}
                >
                  <span>
                    Huron River Drive
                  </span>
                   
                  Shigeto
                </button>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "top songs 0",
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={[Function]}
                >
                  <span>
                    In Da Club
                  </span>
                   
                  50 Cent
                </button>,
                "className": "unselected songs",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      In Da Club
                    </span>,
                    " ",
                    Array [
                      "50 Cent",
                    ],
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "In Da Club",
                    },
                    "ref": null,
                    "rendered": "In Da Club",
                    "type": "span",
                  },
                  " ",
                  "50 Cent",
                ],
                "type": "button",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "top songs 1",
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={[Function]}
                >
                  <span>
                    Cranes In The Sky
                  </span>
                   
                  Solange
                </button>,
                "className": "unselected songs",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Cranes In The Sky
                    </span>,
                    " ",
                    Array [
                      "Solange",
                    ],
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Cranes In The Sky",
                    },
                    "ref": null,
                    "rendered": "Cranes In The Sky",
                    "type": "span",
                  },
                  " ",
                  "Solange",
                ],
                "type": "button",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "top songs 2",
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={[Function]}
                >
                  <span>
                    Huron River Drive
                  </span>
                   
                  Shigeto
                </button>,
                "className": "unselected songs",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Huron River Drive
                    </span>,
                    " ",
                    Array [
                      "Shigeto",
                    ],
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Huron River Drive",
                    },
                    "ref": null,
                    "rendered": "Huron River Drive",
                    "type": "span",
                  },
                  " ",
                  "Shigeto",
                ],
                "type": "button",
              },
              "type": "li",
            },
          ],
          "type": "ol",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
